src/flash/display/Loader.ts

```diff
@@ -83,77 +82,98 @@ module Shumway.AVM2.AS.flash.display {
     };
     static initializer: any = function() {
       var self: Loader = this;
-      DisplayObject._advancableInstances.push(self);
+      display.DisplayObject._advancableInstances.push(self);
     };
 
     static classSymbols: string [] = null;
     static instanceSymbols: string [] = null;
 
-
     /**
-     * Handles the load status and dispatches progress events. This gets manually triggered in the
-     * event loop to ensure the correct order of operations.
+     * In each turn of the event loop, Loader events are processed in two batches:
+     * first INIT and COMPLETE events are dispatched for all active Loaders, then
+     * OPEN and PROGRESS.
+     *
+     * A slightly weird result of this is that INIT and COMPLETE are dispatched one
+     * turn *later* than the other events: INIT is only dispatched if the loadStatus
```

Comment:
In your test case, simple try to create the 2 Loader instances in the second frame of the root timeline instead of the first one. You will see that their `OPEN` and `PROGRESS` events will fire right after `EXIT_FRAME`, withing the same event loop turn, rather then one turn later.